### PR TITLE
[fix]add pull policy

### PR DESCRIPTION
**Pull request by yyf1986**
Signed-off-by: yyf1986 <191557539@qq.com>

the cmd pull  policy is "ifnewer",othe all is "missing".

### CODE REVIEW

**Review by fanux — APPROVED** [review]

File: pkg/image/types/types.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ const (
 	DockerArchive string = "docker-archive"
 )
 
+const PullPolicy = "missing" // processor  create, install
```
Comment (cuisongliu) [inline pkg/image/types/types.go:45]: PullPolicyMissing  ： The meaning is clearer in this way.

File: pkg/image/cmd/save.go [hunk 2]
```diff
@@ -60,5 +61,6 @@ func NewSaveCmd() *cobra.Command {
 	saveCmd.Flags().StringVarP(&archiveName, "output", "o", "", "save image to tar archive file")
 	saveCmd.Flags().StringVarP(&types.DefaultTransport, "transport", "t", types.OCIArchive, fmt.Sprintf("save image transport to tar archive file.(optional value: %s , %s)", types.OCIArchive, types.DockerArchive))
 	saveCmd.Flags().StringVarP(&platform, "platform", "p", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH), "set the OS/ARCH/VARIANT of the image to the provided value instead of the current operating system and architecture of the host (for example linux/arm)")
+	saveCmd.Flags().StringVar(&policy, "policy", "missing", "missing, always, never, ifnewer")
```
Comment (cuisongliu) [inline pkg/image/cmd/save.go:64]: Replace it with a constant PullPolicyMissing.

File: pkg/image/cmd/pull.go [hunk 3]
```diff
@@ -37,9 +38,10 @@ func NewPullCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			return registrySvc.Pull(types.ParsePlatform(platform), args[0])
+			return registrySvc.Pull(types.ParsePlatform(platform), policy, args[0])
 		},
 	}
 	pullCmd.Flags().StringVar(&platform, "platform", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH), "set the OS/ARCH/VARIANT of the image to the provided value instead of the current operating system and architecture of the host (for example linux/arm)")
+	pullCmd.Flags().StringVar(&policy, "policy", "ifnewer", "missing, always, never, ifnewer")
```
Comment (cuisongliu) [inline pkg/image/cmd/pull.go:45]: Replace it with a constant PullPolicyIfNewer.

File: pkg/image/cmd/create.go [hunk 4]
```diff
@@ -70,5 +71,6 @@ with custom cluster name:
 	}
 	createCmd.Flags().StringVarP(&clusterName, "cluster-name", "c", "default", "name of cluster to be created")
 	createCmd.Flags().StringVar(&platform, "platform", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH), "set the OS/ARCH/VARIANT of the image to the provided value instead of the current operating system and architecture of the host (for example linux/arm)")
+	createCmd.Flags().StringVar(&policy, "policy", "missing", "missing, always, never, ifnewer")
```
Comment (cuisongliu) [inline pkg/image/cmd/create.go:74]: Replace it with a constant PullPolicyMissing.

File: pkg/image/cmd/merge.go [hunk 5]
```diff
@@ -49,7 +50,7 @@ merge images:
 	mergeCmd.Flags().StringVarP(&newImageName, "image", "i", "", "image new name")
 	mergeCmd.Flags().StringVar(&options.Platform, "platform", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH), "set the OS/ARCH/VARIANT of the image to the provided value instead of the current operating system and architecture of the host (for example linux/arm)")
 	mergeCmd.Flags().IntVarP(&options.MaxPullProcs, "max-pull-procs", "m", 5, "maximum number of goroutines for pulling")
-
+	mergeCmd.Flags().StringVar(&policy, "policy", "missing", "missing, always, never, ifnewer")
```
Comment (cuisongliu) [inline pkg/image/cmd/merge.go:53]: Replace it with a constant PullPolicyMissing.

**Review by cuisongliu — COMMENTED** [review]
It doesn't seem to be related to modifying the build module?

**Comment by cuisongliu** [comment]
```
--pull string[="true"]          pull the image from the registry if newer or not present in store, if false, only pull the image if not present, if always, pull the image even if the named image is present in store, if never, only use the image present in store if available (default "true")
```

The parameters of build are different from those of pull...

**Comment by yyf1986** [comment]
> It doesn't seem to be related to modifying the build module?

Because it has its own policy。

**Review by cuisongliu — APPROVED** [review]

**Pull request merged**